### PR TITLE
Use `Assertions#assume` in tests

### PR DESCRIPTION
**Pull request by danicheg**
❗ review only with hiding whitespaces applied

### CODE REVIEW

**Comment by danicheg** [comment]
Ahhh, there is a difference in semantics when it comes to linking in JS/Native :( Perhaps it's a reasonable feature request to munit tho.

**Comment by armanbilge** [comment]
I don't think this can be a feature request. This is relying on a compiler feature, that branches can be dead-code-eliminated during the compiler before reaching the JS/Native linker.

Meanwhile, `assume` is a runtime check. It requires that all the code could already be compiled and linked. Unless you make `assume` into a macro ...

**Comment by danicheg** [comment]
> Unless you make assume into a macro ...

yesss

File: tests/shared/src/test/scala/org/http4s/DecodeSpec.scala [hunk 1]
```diff
@@ -93,32 +93,35 @@ class DecodeSpec extends Http4sSuite {
...
-  if (Platform.isJvm)
-    test("decode should handle unmappable character") {
-      // https://stackoverflow.com/a/22902806
-      val source = Stream(0x80.toByte, 0x81.toByte)
-      val decoded =
-        source.through(decodeWithCharset[Fallible](JCharset.forName("IBM1098"))).compile.string
-      val Left(_: UnmappableCharacterException) = decoded
-    }
+  test("decode should handle unmappable character") {
+    assume(Platform.isJvm, "this test is for JVM only")
```
Comment (armanbilge) [inline tests/shared/src/test/scala/org/http4s/DecodeSpec.scala:97]:
```suggestion
    assume(Platform.isJvm, "IBM1098 charset is unavailable on JS/Native")
```

File: tests/shared/src/test/scala/org/http4s/DecodeSpec.scala [hunk 2]
```diff
@@ -93,32 +93,35 @@ class DecodeSpec extends Http4sSuite {
...
-  if (Platform.isJvm)
-    test("decode should not crash in IllegalStateException") {
-      // Found by scalachek
-      val source = Stream(-1.toByte)
-      val decoded =
-        source.through(decodeWithCharset[Fallible](JCharset.forName("x-IBM943"))).compile.string
-      val Left(_: MalformedInputException) = decoded
-    }
+  test("decode should handle overflows") {
+    assume(Platform.isJvm, "this test is for JVM only")
```
Comment (armanbilge) [inline tests/shared/src/test/scala/org/http4s/DecodeSpec.scala:107]:
```suggestion
    assume(Platform.isJvm, "x-ISCII91 charset is unavailable on JS/Native")
```

File: tests/shared/src/test/scala/org/http4s/DecodeSpec.scala [hunk 3]
```diff
@@ -93,32 +93,35 @@ class DecodeSpec extends Http4sSuite {
...
+
+    // Found by scalachek
+    val source = Stream(-36.toByte)
+    val decoded =
+      source.through(decodeWithCharset[Fallible](JCharset.forName("x-ISCII91"))).compile.string
+    assertEquals(decoded, Right("ी"))
+  }
+
+  test("decode should not crash in IllegalStateException") {
+    assume(Platform.isJvm, "this test is for JVM only")
```
Comment (armanbilge) [inline tests/shared/src/test/scala/org/http4s/DecodeSpec.scala:117]:
```suggestion
    assume(Platform.isJvm, "x-ISCII91 charset is unavailable on JS/Native")
```

File: tests/shared/src/test/scala/org/http4s/UriSuite.scala [hunk 4]
```diff
@@ -995,11 +995,12 @@ class UriSuite extends Http4sSuite {
     val ps = Map("param" -> List(1.2, 2.1))
     assertEquals(Uri() =? ps, Uri(query = Query.unsafeFromString("param=1.2&param=2.1")))
   }
-  if (Platform.isJvm || Platform.isNative)
-    test("Uri parameter convenience methods should set a parameter with a float values") {
-      val ps = Map("param" -> List(1.2f, 2.1f))
-      assertEquals(Uri() =? ps, Uri(query = Query.unsafeFromString("param=1.2&param=2.1")))
-    }
+  test("Uri parameter convenience methods should set a parameter with a float values") {
+    assume(Platform.isJvm || Platform.isNative, "this test is for JVM or Native Platforms")
```
Comment (armanbilge) [inline tests/shared/src/test/scala/org/http4s/UriSuite.scala:999]:
```suggestion
    assume(Platform.isJvm || Platform.isNative, "floats on JS have semantics of double")
```

**Review by armanbilge — APPROVED** [review]
LGTM, just a suggestion to add more informative messages.

File: tests/shared/src/test/scala/org/http4s/DecodeSpec.scala (hunk 1 above)
Comment (danicheg) [reply tests/shared/src/test/scala/org/http4s/DecodeSpec.scala:97]: Wow, thanks for that addition. Really helpful info for further explorers 👍🏻

**Pull request merged**